ipaserver/install/server/install.py

```diff
@@ -471,6 +471,15 @@ def install_check(installer):
 
     domain_name = domain_name.lower()
 
+    if options.setup_dns:
+        api.env.domain = domain_name
```

Comment:
This assignment works only because the api hasn't been finalized. It makes me nervous. It makes me wonder if a separate pkg_check() is needed instead of that happening in install_check().

Reply:
me, too.

I'm strongly against this hack. The ``dns.install_check()`` function does a lot of things. Some of the options also depend on a fully working and initialized api object. This needs more engineering work. Perhaps the check can be split into more steps?